### PR TITLE
Fix #1404: failing addIndex with case-sensitive name

### LINKED ISSUE #1404: addIndex seems to fail on names with uppercase (missing double quotes)?
I created a table with the following command:

```
{
  "createTable": {
    "name": "reviews",
    "definition": {
      "primaryKey": "_id",
      "columns": {
        "_id": {
          "type": "text"
        },
        "rating": {
          "type": "int"
        },
        "text": {
          "type": "text"
        },
        "userId": {
          "type": "text"
        },
        "vehicleId": {
          "type": "text"
        },
        "createdAt": {
          "type": "decimal"
        },
        "updatedAt": {
          "type": "decimal"
        }
      }
    }
  }
}
```

Table looks like the following in cqlsh:

image

Running `addIndex` on `vehicleId` property as follows:

```
{
  "addIndex": {
    "column": "vehicleId",
    "indexName": "vehicleId"
  }
}
```

Gives me an error:

```
Error: Command "addIndex" failed with the following errors: [{"message":"Undefined column name vehicleid in table demo.reviews","errorCode":"INVALID_QUERY"}]
```

Likely need to double quote column  names in `addIndex`?

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Fixes issue with name case-sensitivity for "addIndex()" command

**Which issue(s) this PR fixes**:
Fixes #1404

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/AddTableIndexIntegrationTest.java [hunk 1]
```diff
@@ -1,73 +1,68 @@
...
-    @Order(1)
-    public void addIndex() {
-      String json =
-          """
-                      {
-                          "addIndex": {
-                              "column": "age",
-                              "indexName": "age_index"
+    public void addIndexBasic() {
+      DataApiCommandSenders.assertTableCommand(keyspaceName, testTableName)
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/AddTableIndexIntegrationTest.java:41]: Rewritten to use more compact helper method; not changing test itself.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/AddTableIndexIntegrationTest.java [hunk 2]
```diff
@@ -1,73 +1,68 @@
...
+                          "column": "age",
+                          "indexName": "age_index"
+                  }
+                  """)
+          .hasNoErrors()
+          .body("status.ok", is(1));
+    }
+
+    @Test
+    public void addIndexCaseSensitive() {
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/AddTableIndexIntegrationTest.java:55]: And this is the new test.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/AddTableIndexIntegrationTest.java [hunk 3]
```diff
@@ -1,73 +1,68 @@
...
-                                              "type": "text"
-                                          }
-                                      },
-                                      "primaryKey": "id"
-                                  }
-                          }
-                    """
-            .formatted(simpleTableName);
-    createTable(tableJson);
+    createTableWithColumns(
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/AddTableIndexIntegrationTest.java:21]: Using more-compact helper method but also adding case-sensitive column "vehicleId"

**Comment by amorton** [comment]
addIndex is still POC - can we hold off adding any new tests for it until we have re-worked it to beta quality ?

**Comment by tatu-at-datastax** [comment]
> addIndex is still POC - can we hold off adding any new tests for it until we have re-worked it to beta quality ?

The whole API Tables is PoC... ? :-)

We already had a test and the only change is to ensure case-sensitivity is retained so this is not a big change

But since @vkarpov15 reported this, it might be blocking him -- this is the main reason I was thinking of fixing this.

Anyway I am fine either way, we can discuss.

**Review by vkarpov15 — COMMENTED** [review]
LGTM. I wouldn't say this is a blocker, I worked around it by changing variable names I need to add indexes on to snake_case, but I figure this is an easy enough fix.

**Comment by tatu-at-datastax** [comment]
Thank you @vkarpov15 . We'll discuss this today in team meeting.

**Review by maheshrajamani — APPROVED** [review]

**Comment by vkarpov15** [comment]
Confirmed this fix works locally :+1:

**Pull request merged**